include/mbedtls/ssl.h

```diff
@@ -3851,25 +3851,35 @@ int mbedtls_ssl_renegotiate( mbedtls_ssl_context *ssl );
 int mbedtls_ssl_read( mbedtls_ssl_context *ssl, unsigned char *buf, size_t len );
 
 /**
- * \brief          Try to write exactly 'len' application data bytes
+ * \brief          Queue application data for dispatch
  *
- * \warning        This function will do partial writes in some cases. If the
- *                 return value is non-negative but less than length, the
- *                 function must be called again with updated arguments:
- *                 buf + ret, len - ret (if ret is the return value) until
- *                 it returns a value equal to the last 'len' argument.
+ * \warning        This function may queue less data for dispatch than requested.
+ *                 If the return value is non-negative but less than \p len,
+ *                 the function should be called again for the remaining data.
  *
```

Comment:
Please explain the blocking behavior of this function. In particular, is it blocking when the TLS socket is in blocking mode?